gtk/src/subclass/entry.rs

```diff
@@ -0,0 +1,83 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::Cast;
+
+use super::widget::WidgetImpl;
+use crate::Entry;
+use crate::Widget;
+
+pub trait EntryImpl: EntryImplExt + WidgetImpl {
+    fn populate_popup(&self, entry: &Self::Type, popup: &Widget) {
+        self.parent_populate_popup(entry, popup)
+    }
+
+    fn activate(&self, entry: &Self::Type) {
+        self.parent_activate(entry)
+    }
+}
+
+pub trait EntryImplExt: ObjectSubclass {
+    fn parent_populate_popup(&self, entry: &Self::Type);
+    fn parent_activate(&self, entry: &Self::Type);
+}
+
+impl<T: EntryImpl> EntryImplExt for T {
+    fn parent_populate_popup(&self, entry: &Self::Type, popup: &Widget) {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().parent_class() as *mut ffi::GtkEntryClass;
+            if let Some(f) = (*parent_class).populate_popup {
+                f(
+                    entry.unsafe_cast_ref::<Entry>().to_glib_none().0,
+                    popup.to_glib_none().0,
+                )
+            }
+        }
+    }
+
+    fn parent_activate(&self, entry: &Self::Type) {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().parent_class() as *mut ffi::GtkEntryClass;
+            if let Some(f) = (*parent_class).activate {
+                f(entry.unsafe_cast_ref::<Entry>().to_glib_none().0)
+            }
+        }
+    }
+}
+
+unsafe impl<T: EntryImpl> IsSubclassable<T> for Entry {
+    fn class_init(class: &mut glib::Class<Self>) {
+        Self::parent_class_init::<T>(class);
+
+        if !crate::rt::is_initialized() {
```

Comment:
Isn't it checked in the parent class already?

Reply:
I just blindly copied from button :)

Reply:
All the other modules seem to do the same, but it is true that this is checked in widget.rs... Maybe we should drop it everywhere, but in that case it should go in a separate MR

Reply:
Sounds like a plan. I'll send a PR for that.

Reply:
Maybe first `git blame` why they were added... there's a reason.

Reply:
I was the one adding this check, just didn't think it through: https://github.com/gtk-rs/gtk3-rs/commit/cb53101568b1332acb48494ab5d82588bc04bd77

Reply:
Apparently you didn't :) See https://github.com/gtk-rs/gtk3-rs/issues/543

I think this code is correct and needed.

Reply:
The issue is apparently about non-widget items. Oh well...

Reply:
@bilelmoussaoui can probably tell us for sure

Reply:
I did that to panic as early as possible

Reply:
Shouldn't it need to be before `Self::parent_class_init::<T>(class);` then?

Reply:
I see you opened a PR, so I will resolve the conversation here for now.